include/envoy/http/filter.h

```diff
@@ -596,6 +608,36 @@ class StreamFilterBase {
    * @param action the resulting match action
    */
   virtual void onMatchCallback(const Matcher::Action&) {}
+
+  struct LocalReplyData {
+    // The error code which (barring reset) will be sent to the client.
+    Http::Code code_;
+    // The details of why a local reply is being sent.
+    absl::string_view details_;
+    // True if a reset will occur rather than the local reply (some prior filter
+    // has returned ContinueAndResetStream)
+    bool reset_imminent_;
+  };
+
+  /**
+   * Called after sendLocalReply is called, and before any local reply is
+   * serialized either to filters, or downstream.
+   * This will be called on both encoder and decoder filters starting at the
+   * router filter and working towards the first filter configured.
```

Comment:
In theory someone could be using another terminal filter that isn't the router filter and still make use of local replies right? Maybe make this "at the terminal filter"